plugins/optimization-detective/tests/class-optimization-detective-test-helpers.php

```diff
@@ -166,6 +166,35 @@ function ( array $element ) use ( $params ): array {
 		return new OD_URL_Metric( $data );
 	}
 
+	/**
+	 * Stores URL Metric by merging it with the other URL Metrics which share the same normalized query vars.
+	 *
+	 * @param non-empty-string $slug Slug (hash of normalized query vars).
+	 * @param OD_URL_Metric    $new_url_metric New URL Metric.
+	 * @return positive-int|WP_Error Post ID on success, or WP_Error on failure.
+	 */
+	public function store_url_metric( string $slug, OD_URL_Metric $new_url_metric ) {
+		$post = OD_URL_Metrics_Post_Type::get_post( $slug );
+		$etag = $new_url_metric->get_etag();
+
+		$group_collection = new OD_URL_Metric_Group_Collection(
+			$post instanceof WP_Post ? OD_URL_Metrics_Post_Type::get_url_metrics_from_post( $post ) : array(),
+			$etag,
+			od_get_breakpoint_max_widths(),
+			od_get_url_metrics_breakpoint_sample_size(),
+			od_get_url_metric_freshness_ttl()
+		);
+
+		try {
+			$group = $group_collection->get_group_for_viewport_width( $new_url_metric->get_viewport_width() );
+			$group->add_url_metric( $new_url_metric );
+		} catch ( InvalidArgumentException $e ) {
+			return new WP_Error( 'invalid_url_metric', $e->getMessage() );
+		}
```

Comment:
I think this can be further simplified:

```suggestion
		$group_collection->add_url_metric( $new_url_metric );
```

Note that no exception should actually ever be thrown because the URL Metric's viewport width will _always_ be within the range of zero to infinity. And especially in the test case helper method, we don't need to add the exception handling since if an error occurs we would want the test to fail terribly since this should never happen.

Reply:
I realize that this is what was originally in `store_url_metric`, but I think it was wrong there too.